src/Core/Foundation/Version/Version.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Foundation\Version;
+
+use PrestaShop\PrestaShop\Core\Foundation\Version\Exception\InvalidVersionException;
+
+/**
+ * Class responsible of managing the right version of Shop
+ * for every internal/external services.
+ */
+class Version
+{
+    const STRING = 0;
+    const INTEGER = 1;
+
+    /**
+     * Full version name.
+     *
+     * @var string
+     */
+    private $fullVersion;
+
+    /**
+     * Major version name.
+     *
+     * @var string
+     */
+    private $majorVersionString;
+
+    /**
+     * Major version.
+     *
+     * @var int
+     */
+    private $majorVersion;
+
+    /**
+     * Minor version.
+     *
+     * @var int
+     */
+    private $minorVersion;
+
+    /**
+     * Release version.
+     *
+     * @var int
+     */
+    private $releaseVersion;
+
+
+    public function __construct($version, $majorVersionString, $majorVersion, $minorVersion, $releaseVersion)
```

Comment:
wouldnt it be better to pass single full version like `'1.7.4.0'` and then parse different formats instead of passing a lot of versions? @mickaelandrieu what do you think?

Reply:
@sarjon @PierreRambaud said the same above. I just did it like Symfony does it without knowing the reason. Anyway I like it like it is and Symfony will have some goog reason. If you change it in another PR it's also ok for me of course.